sklearn/ensemble/weight_boosting.py

```diff
@@ -576,9 +582,10 @@ def _boost_discrete(self, iboost, X, y, sample_weight, random_state):
         # Only boost the weights if I will fit again
         if not iboost == self.n_estimators - 1:
             # Only boost positive weights
-            sample_weight *= np.exp(estimator_weight * incorrect *
-                                    ((sample_weight > 0) |
-                                     (estimator_weight < 0)))
+            sample_weight = np.exp(np.log(sample_weight) +
```

Comment:
Let's include a comment here explaining why we are doing this in log space.

Reply:
Not sure if we had an advantage doing this we still had an overflow.